flow/project.py

```diff
@@ -914,25 +1233,61 @@ def _generate_id(self, job, operation_name=None, index=0):
         if max_len < len(job_op_id):
             raise ValueError("Value for MAX_LEN_ID is too small ({}).".format(self.MAX_LEN_ID))
 
+        if len(jobs) > 1:
+            aggregate_id = str(jobs[0])[:8]+'-'+str(jobs[-1])[:8]
+        else:
+            aggregate_id = str(jobs[0])[:8]
+
         separator = getattr(project._environment, 'JOB_ID_SEPARATOR', '/')
-        readable_name = '{project}{sep}{job}{sep}{op_string}{sep}{index:04d}{sep}'.format(
-                    sep=separator,
-                    project=str(project)[:12],
-                    job=str(job)[:8],
-                    op_string=op_string[:12],
-                    index=index)[:max_len]
+        readable_name = '{project}{sep}{job}{sep}{len}{sep}{group}{sep}' \
+                        '{index:04d}{sep}'.format(
+                            sep=separator,
+                            project=str(project)[:12],
+                            job=aggregate_id,
+                            len=len(jobs),
+                            group=self.name,
+                            index=index)[:max_len]
 
         # By appending the unique job_op_id, we ensure that each id is truly unique.
         return readable_name + job_op_id
 
-    def _get_status(self, job):
-        """For a given job check the groups submission status."""
+    def _get_status(self, jobs):
+        """For given jobs check the groups submission status."""
         try:
-            return JobStatus(job._project.document['_status'][self._generate_id(job)])
+            return JobStatus(jobs[0]._project.document['_status'][self._generate_id(jobs)])
         except KeyError:
             return JobStatus.unknown
 
-    def _create_submission_job_operation(self, entrypoint, default_directives, job,
+    def _get_filtered_jobs(self, jobs):
+        "Returns filtered jobs for this group"
+        jobs = list(jobs)
+        filter = self.flow_select
+        grouper, sort = self.flow_aggregate
+        jobs_list = filter(jobs)
+        if sort is not None:
+            jobs_list = sort(jobs_list)
+        jobs_list = grouper([job for job in jobs_list])
+        jobs_list = [[j for j in jobs] for jobs in jobs_list]
+        if not len(jobs_list):
+            return None
+        for i, job in enumerate(jobs_list[-1]):
+            if job is None:
+                del jobs_list[-1][i:]
+                break
+        return jobs_list
```

Comment:
This should be done largely using an `Aggregate` class or an external function at least. The filtering is fine.